provider/internal/java/dependency.go

```diff
@@ -152,7 +158,10 @@ func (p *javaServiceClient) GetDependenciesDAG() ([]provider.DepDAGItem, uri.URI
 	}
 	filepath.WalkDir(moddir, w.walkDirForJar)
```

Comment:
I think all .jar files here will be incorrectly associated with the pom file `file`. This was always a problem I think. I am fine with addressing this later though. We need two fixes here: 1. Change file URI for JARs 2. Look for JARs in fallback parser too

Reply:
Yeah I agree I can make this change fairly quickly I think so I think we should just do it on this PR

Reply:
@shawn-hurley thanks, I had already created an issue to capture that work, I will link it with the PR

Reply:
Perfect pretty sure the change should be there now